src/main/java/no/digipost/signature/client/core/internal/ClientHelper.java

```diff
@@ -96,30 +92,31 @@ public void cancel(Cancellable cancellable) {
             String url = cancellable.getCancellationUrl().getUrl();
             Response response = postEmptyEntity(url);
             Status status = Status.fromStatusCode(response.getStatus());
-            if (status == CONFLICT) {
+            if (status == OK) {
+                return;
+            } else if (status == CONFLICT) {
                 throw new NotCancellableException();
             }
-            handleGeneralError(response, status);
+            throw handleGeneralError(response, status);
         } else {
             throw new NotCancellableException();
```

Comment:
Valid point :)

The `Cancellable` is «not cancellable» in cases where the client receives an updated portal job status after a poll and it is `completed`, i.e. signed or rejected by all signers.

While the cause of the two different cases you point out is *probably* the same (the job has been completed), I agree that a way to distinguish the two is a good idea. I'll look into it.